tests/unit/model/test_element.py

```diff
@@ -53,3 +54,40 @@ def test_model_reference():
     element = ConcreteElement(name="Element")
     element.set_model(model)
     assert element.get_model() is model
+
+
+def test_element_can_only_add_relationship_to_source():
+    """Make sure that nothing adds a relationship to the wrong element."""
+    element1 = ConcreteElement(name="elt1")
+    element2 = ConcreteElement(name="elt1")
+    with pytest.raises(
+        ValueError,
+        match="Cannot add relationship .* to element .* that is not its source",
+    ):
+        element1.add_relationship(source=element2)
+
+
+def test_element_add_relationship_can_omit_source():
+    """
+    When using add_relationship with params rather than an existing instance,
+    it should default the source.
+    """
```

Comment:
It's a bit pedantic but since we plan to use flake8-docstrings in future, the first line in a docstring should fit on one line and be imperative, for example,

```suggestion
    """Expect that creating a relationship uses the default source."""
```

Reply:
If there is more to say you can add more paragraphs to the docstring.

Reply:
I'm good with pedantic.

Reply:
I've sorted this one and also some others that flake8 was showing around the files in this PR.  More still to do though.